src/tokenize.c

```diff
@@ -149,9 +149,11 @@ static long long parse_integer(const char *str, Location location, int nbits)
         base = 2;
         digits = &str[2];
         valid_digits = "01";
-    } else if (str[0] == '0' && str[1] != '\0') {
-        // 0777 in C actually means 511. Jou does not allow writing 0777.
-        fail_with_error(location, "unnecessary zero at start of number");
+    } else if (str[0] == '0' && str[1] == 'o') {
+        // 0o777

```

Comment:
Can you add a test file to `tests/should_succeed/`? It could simply print a few numbers that use the `0o...` syntax. See `tests/should_succeed/expfloat.jou` for example.

Reply:
I will add

Reply:
```
from "stdlib/io.jou" import printf

def main()-> int:
    printf("%d", 0o777) # Should be 511
    printf("%d", 0o337522) # Should be 114514
    printf("%d", 0o7244314) # Should be 1919180
    return 0
```